### PR TITLE
Inherit Base and add logging to backup/restore/upgrade

### PR DESCRIPTION
**Pull request by MaximilianMeister**

### CODE REVIEW

**Comment by MaximilianMeister** [comment]
rebased

**Comment by rhafer** [comment]
lgtm, but needs a rebase.

**Comment by MaximilianMeister** [comment]
@rhafer rebased

**Comment by rhafer** [comment]
+1

**Comment by aplanas** [comment]
+1

File: crowbar_framework/lib/crowbar/upgrade.rb [hunk 1]
```diff
@@ -17,7 +17,7 @@
 require "tempfile"
 
 module Crowbar
-  class Upgrade
+  class Upgrade < Backup::Base
```
Comment (rhafer) [inline crowbar_framework/lib/crowbar/upgrade.rb]: Doesn't work. ```NameError (uninitialized constant Backup::Base)```

Comment (rhafer) [inline crowbar_framework/lib/crowbar/upgrade.rb]: Maybe ```Crowbar::Backup::Base``` ?

Comment (MaximilianMeister) [inline crowbar_framework/lib/crowbar/upgrade.rb]: right, maybe because Crowbar is a module it doesn't work...

**Comment by rhafer** [comment]
+1 again

**Comment by aplanas** [comment]
+1

File: crowbar_framework/lib/crowbar/upgrade.rb [hunk 2]
```diff
@@ -17,7 +17,7 @@
 require "tempfile"
 
 module Crowbar
-  class Upgrade
+  class Upgrade < Crowbar::Backup::Base
```
Comment (tboerger) [inline crowbar_framework/lib/crowbar/upgrade.rb]: Upgrade extends from Backup base class? Sounds simply weird...

**Pull request merged**